src/components/input/input.stories.mdx

```diff
@@ -1,8 +1,16 @@
 import { Story, Preview, Meta } from '@storybook/addon-docs/blocks';
-import { withKnobs, text, boolean, select } from '@storybook/addon-knobs';
+import {
+  withKnobs,
+  text,
+  number,
+  boolean,
+  select,
+} from '@storybook/addon-knobs';
+import { useEffect } from '@storybook/client-api';
 import textDemo from './demo/example.twig';
 import selectDemo from './demo/select.twig';
 import './demo/styles.scss';
+import { createElasticTextArea } from './input.ts';
```

Comment:
I was expecting the filename to more closely be related to the module itself. Might we consider something like:
```js
import { createElasticTextArea } from './elastic-textarea.ts`;
```
🤔

Reply:
Interesting question!

I was operating on the assumption that "Input" is the component, and this is the component-specific JS, one function of which is `createElasticTextArea`.

If we added another feature to the Input component, would you expect it to live in a separate file or the same?

Reply:
I'd prefer these to be split up into separate files. That should make it easier to only import the functionality you actually need on your page.

Side question: would this work on a naked `<textarea>` or does it rely on the component CSS?

Reply:
Excellent question, @spaceninja. 🙂 

> If we added another feature to the Input component, would you expect it to live in a separate file or the same?

I would imagine each module corresponds to one feature. I would like to avoid massive TS files if possible.

I'm not sure how this would be packaged up, though, for consumer use. Would there be one `input/index.ts` file that re-exports the `elastic-textarea` module? Something like:
- `input/elastic-textarea.ts`
- `input/another-input-feature.ts`
- `input/the-perfect-select.ts`
- `input/index.ts`

Or is it something like?
- `elastic-textarea/index.ts`
- `another-input-feature/index.ts`
- `the-perfect-select/index.ts`

I suppose this is more an architectural question. I'm curious what the thoughts of @Paul-Hebert and @calebeby are. I'm also curious what open-source projects might be doing. 🔍

Reply:
That's fine, knowing we prefer smaller, feature-specific files makes sense.

To answer your other questions, there's no need for an index file (I dislike index files anyway because it makes it difficult to tell which file you're editing when you have multiple `index.js` files open)

Any `*.ts` files in the component directory will be added to the compiled package library JS file, and we'll just need to import each file as needed into the stories that need them (which I think is fine)

Assuming @Paul-Hebert and @calebeby have no objections, I'll rename this file.

TAKEAWAY: If a component has multiple JS features, each should live in a separate, clearly-named `.ts` file.

Reply:
> Side question: would this work on a naked <textarea> or does it rely on the component CSS?

It relies on the component CSS to set `height:auto`, which I've added to the documentation.

And we added a `maxRows` last night to avoid triggering an infinite loop if the users forgets that class or if the textarea has a fixed height for any reason.

Reply:
I'd also prefer to leave out the index file unless strictly necessary. I can imagine a really complex feature having its own directory with multiple modules and an index file, but for this example I'd rather have `input/elastic-textarea.ts`

Reply:
File renamed! I'm putting this back up for review now.